advice_animal/runner.py

```diff
@@ -11,10 +15,162 @@
+            cur_cwd = os.getcwd()
+            try:
+                for advice_name, check_cls in self.iter_check_classes(
+                    confidence_filter, preview_filter, name_filter
+                ):
+                    LOG.log(VLOG_1, "Running check %s", advice_name)
+                    env = Env(repo)
+                    for project in env.py_projects:
+                        os.chdir(project)
+                        try:
+                            check = check_cls(env)
+                            changes_needed = bool(check.check())
+                            results[advice_name] = Result(
+                                advice_name=advice_name,
+                                project=project,
+                                success=True,
+                                changes_needed=changes_needed,
+                            )
+                        except Exception as e:
+                            results[advice_name] = Result(
+                                advice_name=advice_name,
+                                project=project,
+                                success=False,
+                                message=str(e),
+                            )
+                            return results
+            finally:
+                os.chdir(cur_cwd)
+        else:
+            cur_cwd = os.getcwd()
+            with tempfile.TemporaryDirectory() as d:
+                run_cmd(["git", "clone", repo, d])
+                os.chdir(d)
+                env = Env(Path(d))
+                try:
+                    for advice_name, check_cls in self.iter_check_classes(
+                        confidence_filter, preview_filter, name_filter
+                    ):
+                        try:
+                            output = ""
+                            changes_needed = False
+                            LOG.log(VLOG_1, "Running check %s", advice_name)
+                            run_cmd(
+                                [
+                                    "git",
+                                    "checkout",
+                                    "-b",
+                                    advice_name,
+                                    f"origin/{current_branch}",
+                                ]
+                            )
+                            for project in env.py_projects:
+                                LOG.log(VLOG_1, "Checking %s", project)
+                                os.chdir(project)
+                                check = check_cls(env)
+                                changes_needed = bool(check.check())
+                                run_cmd(["git", "add", "-A"])
+                            if self.mode == "apply":
+                                run_cmd(["git", "commit", "-m", f"Apply {advice_name}"])
+                                run_cmd(["git", "push", "-f", "origin", advice_name])
```

Comment:
I think the old one prefixed branch names with "advice-" rather than using the name directly.  I like that and intended it to be something configurable in the future.